lit_llama/adapter.py

```diff
@@ -150,9 +150,9 @@ def forward(
             if adapter_kv_cache is not None:
                 ak, av = adapter_kv_cache # 2 * (B, nh, aT, hs)
             else:
-                prefix = self.adapter_wte.weight.reshape(1, self.adapter_prompt_length, self.n_embd)
-                aT = prefix.size(1)
-                _, ak, av = self.c_attn(prefix).split(self.n_embd, dim=2) # (1, aT, 3 * C) --> 3 * (1, aT, C)
+                aT = self.adapter_prompt_length
```

Comment:
If you pull this out of the `if`, then in L162 you can do `torch.ones(T, aT, ...)` which would look nicer than accessing the shapes

Reply:
Thanks for the suggestion 👍 
Totally agree. 
Plus it helps with line 151. Every time I looked at the shapes in this line in my head I heard "referenced before assignment" 😄 .